src/components/NewPoll/NewPollForm.js

```diff
@@ -1,101 +1,68 @@
-import React, { useState, useEffect, useCallback } from "react";
+import React, { useEffect } from "react";
+import { connect } from "react-redux";
+import * as messageAction from "../../redux/messageActions";
+import * as channelAction from "../../redux/channelActions";
+
 import styled from "styled-components";
 import colours from "../../assets/colours";
+
+import { numberParser } from "../shared/helperFunctions";
 import { SelectionOfChannels } from "./SelectionOfChannels";
-import { useHistory } from "react-router-dom";
-import { ReactComponent as Trash } from "../../assets/trash.svg";
-import { useSelector, useDispatch } from "react-redux"; // hooks provided by react-redux;
-import * as actions from "../../redux/actions";
-import { Link } from "react-router-dom";
-import { API_BASE_URL } from "../../backend-request/index";
-
-export const NewPollForm = () => {
-  const history = useHistory();
-  const [channels, setChannels] = useState([]);
-  const [error, setError] = useState(false);
-
-  //Get your data from the store
-  const question = useSelector(state => state.question);
-  const responses = useSelector(state => state.responses);
-  const channel = useSelector(state => state.channel);
-  const channelID = useSelector(state => state.channelID);
-  const channelSize = useSelector(state => state.channelSize);
-
-  //dispatch changes to store.
-  const dispatch = useDispatch();
-
-  const updateQuestion = newQuestion =>
-    dispatch(actions.updateQuestion(newQuestion));
-
-  // using useCallback from react to wrap useDispatch of react-redux , useCallback is an overkill for now , but if we are to pass updateAnswers to a child component it is better to wrap it as given to prevent child component rerendering
-  const updateAnswers = newAnswer =>
-    dispatch(actions.updateResponses(newAnswer));
-
-  const updateChannel = newChannel =>
-    dispatch(actions.updateChannel(newChannel));
-
-  const updateChannelID = newChannelID =>
-    dispatch(actions.updateChannelID(newChannelID));
-
-  const updateChannelSize = newChannelSize =>
-    dispatch(actions.updateChannelSize(newChannelSize));
-
-  const handleSubmitPoll = event => {
-    event.preventDefault();
-
-    let isResponseValid = responses.length >= 2;
-    responses.forEach(response => {
-      if (isResponseValid && response == "") {
-        isResponseValid = false;
-      }
-    });
-
-    // setState();
-    if (question == "" || !isResponseValid || channel == "") {
-      // but you can use a location instead
-      setError(true);
-      console.log("Make an error message for each box below");
-    } else {
-      setError(false);
-      history.push("/poll-submitted");
-    }
+import Response from "./Response";
+
+const NewPollForm = ({
+  updateQuestion,
+  updateAnswers,
+  updateChannel,
+  updateChannelID,
+  updateChannelSize,
+  message,
+  fetchChannels,
+  channels
+}) => {
+  useEffect(() => {
+    fetchChannels();
+  }, [fetchChannels]);
```

Comment:
Why pass `fetchChannels` in the array? `useEffect` uses that array to determine when the effect needs to be re-run. If anything in the array changes, the effect gets re-run. But...from what I can see, `fetchChannels` will never change, so you could just pass `[]` here.

Reply:
fixed